Networking/Networking/Model/CreateBlazeCampaign.swift

```diff
@@ -0,0 +1,138 @@
+import Foundation
+import Codegen
+
+/// Represents the entity sent for creating a new Blaze Campaign
+///
+public struct CreateBlazeCampaign: Encodable, Equatable, GeneratedFakeable, GeneratedCopiable {
+    public struct Image: Encodable, Equatable, GeneratedFakeable, GeneratedCopiable {
+        /// URL of the image for the campaign
+        ///
+        public let url: String
+
+        /// mime type of the image
+        ///
+        public let mimeType: String
+
+        public init(url: String, mimeType: String) {
+            self.url = url
+            self.mimeType = mimeType
+        }
+    }
+
+    public struct Targeting: Encodable, Equatable, GeneratedFakeable, GeneratedCopiable {
+        /// IDs of locations => GET /locations/search?keyword=city
+        ///
+        public let locations: [Int64]?
+
+        /// IDs of languages => GET /targeting/languages
+        ///
+        public let languages: [String]?
+
+        /// IDs of devices => GET /targeting/devices (absent if any)
+        ///
+        public let devices: [String]?
+
+        /// IDs of pageTopics => GET /targeting/page-topics (absent if any)
+        ///
+        public let pageTopics: [String]?
+
+        public init(locations: [Int64]?, languages: [String]?, devices: [String]?, pageTopics: [String]?) {
+            self.locations = locations
+            self.languages = languages
+            self.devices = devices
+            self.pageTopics = pageTopics
+        }
+    }
+
+    /// Origin of the campaign creation
+    /// Accepted options are calypso, jetpack, wc-mobile-app, wp-mobile-app
+    ///
+    public let origin: String
+
+    /// See peeHDf-2ii-p2
+    ///
+    public let paymentMethodID: String
+
+    /// Sample format `2023-12-05`
+    /// Including, can't be less than 24 hours after submission time
+    ///
+    public let startDate: String
+
+    /// Sample format `2023-12-11`
+    /// Including
+    ///
+    public let endDate: String
```

Comment:
For simplicity, we should use `Date` and send `dateFormatter` to the `toDictionary` method when converting the object like how we did with `BlazeForecastedImpressionsInput`.

Reply:
Updated as part of https://github.com/woocommerce/woocommerce-ios/pull/11617/commits/e261f537d0c791f4d9a77f65122b26c1a9ca28d6